src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -769,15 +769,26 @@ def is_weight_compression_supported(
 
         return is_supported_dtype and not no_bit_reduction
 
-    def apply(
+    def get_processed_weight_compression_parameters(
         self,
         model: TModel,
         graph: NNCFGraph,
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
-    ) -> TModel:
-        self.set_backend_entity(model)
+    ) -> tuple[list[WeightCompressionParameters], list[WeightCompressionParameters], dict[str, Any]]:
+        """
+        Collects and processes weight compression parameters for all nodes in the model,
+        applies all processing steps including mixed precision assignment, group size fallback
+        handling and compression configuration setting.
 
+        :param model: Backend-specific input model.
+        :param graph: NNCFGraph instance.
+        :param statistic_points: Optional pre-collected statistic points.
+        :param dataset: Optional dataset for statistics collection.
+        :return: A tuple containing Final processed list of weight parameters ready for compression,
+            List of weight parameters that were skipped
+            Statistics collected for data-aware compression. None if data-free compression.
```

Comment:
```suggestion
            statistics collected for data-aware compression. None if data-free compression.
```
Does None related to the last element in return tuple? It is not clear